install/kubernetes/cilium/templates/cilium-nodeinit-daemonset.yaml

```diff
@@ -93,9 +93,6 @@ spec:
                       while PATH="${PATH}:/home/kubernetes/bin" crictl ps | grep -v "node-init" | grep -q "POD_cilium"; do sleep 1; done
                     fi
 
-                    systemctl disable sys-fs-bpf.mount || true
```

Comment:
@aditighag you had mentioned that this may resolve https://github.com/cilium/cilium/issues/15393 , but this file still performs `nsenter` with a script that expects to run bash via the `#!/bin/bash` line at the start of the script. I don't think this PR will solve bottlerocket deployments just yet.

Reply:
`nsenter` with mount as the enterable namespace was required to mount BPF fs. If this PR introduces bidirectional mount as an alternative (L142-179 are removed in this PR), then I don't see any reason to run the script using `nsenter` with enterable mount namespace anymore.

@aanm You'll need to make changes to this script - https://github.com/cilium/image-tools/blob/885b789b31a535dd5a567242c8161971d6b34387/images/startup-script/manage-startup-script.sh#L60. 

Also, update the note @L130 - 

```
# STARTUP_SCRIPT is the script run on node bootstrap. Node
          # bootstrapping can be customized in this script. This script is invoked
          # using nsenter, so it runs in the host's network and mount namespace using
          # the host's userland tools!
```

Reply:
@aditighag `nsenter` is not only used for the bpf fs, it is used to access all files modified by the script.

The node-init script / daemonset was initially designed for gke and later on for aks and eks. All of these environments work with the script since we have CI coverage for them, for all other environments either Cilium works out-of-the-box or some other changes need to be added but they are definitely out of scope of this PR.